src/Neo/Network/P2P/RemoteNode.cs

```diff
@@ -81,8 +81,8 @@ public RemoteNode(NeoSystem system, LocalNode localNode, object connection, IPEn
         {
             this.system = system;
             this.localNode = localNode;
-            knownHashes = new HashSetCache<UInt256>(system.MemPool.Capacity * 2 / 5);
-            sentHashes = new HashSetCache<UInt256>(system.MemPool.Capacity * 2 / 5);
+            _knownHashes = new HashSetCache<UInt256>(Math.Min(1_000, system.MemPool.Capacity * 2 / 5));
+            _sentHashes = new HashSetCache<UInt256>(Math.Min(1_000, system.MemPool.Capacity * 2 / 5));
```

Comment:
That's the bug

Reply:
Not sure how it'll behave in cases similar to #2188.

Reply:
Your tool simulate remote nodes to send the txs or it works without them?

Reply:
We're sending transactions, how would we count TPS otherwise? It's a regular network with CNs and RPC, can be C# for everything, can be NeoGo, can be a mix.

Reply:
You can send them by rpc without nodes connected

Reply:
In a multi-node network there is always some P2P traffic. But even in a single-node scenario we're using a separate RPC node, transactions get into CN via P2P.